### ISSUE TITLE
Add ignoreTypes: [] option to selector-no-type

### ISSUE THREAD

**Issue by Garbee** [stylelint/stylelint#1706]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

This is a feature request for `selector-no-type` to be allowed a whitelist of types to be used.

> Which rule, if any, is this issue related to?

`selector-no-type`

> What CSS is needed to reproduce this issue?

e.g.

```css
fieldset:disabled .a-button {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```yaml
extends: stylelint-config-standard
rules:
  selector-no-type: true
```

> Which version of stylelint are you using?

`7.0.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Automated through webpack.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Not initially, but we are using Scss so our source looks more like `fieldset:disabled &` over having the selector directly behind the type.

> What did you expect to happen?

I expect a configuration option to allow us to specifically allow `fieldset` (or any other given elements to the whitelist) to not trigger the `unexpected type error`.

> What actually happened (e.g. what warnings or errors you are getting)?

Unexpected type error on the line defining the disabled fieldset target.

**Comment by evilebottnawi** [comment]
sgtm add options `ignoreTypes`

**Comment by jeddy3** [comment]
@stylelint/core Any objections to, as @evilebottnawi suggests, adding an `ignoreTypes: []` secondary option?

**Comment by Garbee** [comment]
It will be a recurring thing with the library I'm working on. As we are building multiple components and a few form-related pieces (checkbox, radio, button, etc.) will all need to have this in them. It would be far better if we have some global way of saying `fieldset` is allowed so we don't need to remember to do an inline comment each time.

**Comment by davidtheclark** [comment]
`ignoreTypes: []` seems fine to me.

**Comment by jeddy3** [comment]
> It would be far better if we have some global way of saying `fieldset` is allowed so we don't need to remember to do an inline comment each time.

Thanks for the explanation. It looks like everyone is fine with a new secondary option. So, I'll label this up as "help wanted".

@Garbee Would you like to contribute this option? We recently added an `ignoreTypes: []` secondary option to [`selector-type-no-unknown`](https://github.com/stylelint/stylelint/tree/master/src/rules/selector-type-no-unknown). You'll find everything you need there: [code](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-type-no-unknown/index.js#L70), [tests](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-type-no-unknown/__tests__/index.js#L121-L149) and [docs](https://github.com/stylelint/stylelint/tree/master/src/rules/selector-type-no-unknown#optional-options). Step-by-step instructions for adding a new option are in the [developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules), along with how to [run tests](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#running-tests).

**Comment by Garbee** [comment]
Sure, I'll take a look at making a PR tomorrow for this.
